### ISSUE TITLE
Python Errors during Dataset Upload: ConnectionError and JSONDecodeError

### ISSUE THREAD

**Issue by artyaltanzaya** [roboflow/roboflow-python#163]
#### Environment:
- **OS:** macOS with M1 Max
- **Python Code:** 
  ```python
  my_workspace.upload_dataset(dataset_path=public_dataset.location, num_workers=100, project_name="protectiveequipment4")
  ```

#### Description:
While trying to upload a large dataset using the provided Python code snippet, I encountered two different errors:

1. **ConnectionError:**
   ```
   ConnectionError: HTTPSConnectionPool(host='api. roboflow. com', port=443): Max retries exceeded with url: /dataset/protectiveequipment-8ugbr/u
   ```
   Fixed this by setting the `CURL_CA_BUNDLE` environment variable to an empty string:
   ```python
   import os
   os.environ['CURL_CA_BUNDLE'] = ''
   ```
   After this fix, I encountered the second error.

2. **JSONDecodeError:**
   ```
   JSONDecodeError: Expecting value: line 1 column 1 (char 0)
   ```

#### Steps to Reproduce:
1. Run the provided Python code snippet to attempt dataset upload.
2. Observe the `ConnectionError`, and fix it using the provided workaround.
3. Observe the `JSONDecodeError`.

#### Expected Results:
The Python code should successfully upload the dataset without any errors.

#### Actual Results:
Two distinct errors are encountered: a `ConnectionError` that can be fixed with a workaround, and a `JSONDecodeError` that needs further investigation.

#### Additional Context:
The most number of images I was able to upload via the API was around 5000.

#### Attachments:

![Screenshot 2023-08-07 at 12 04 02 PM](https://github.com/roboflow/roboflow-python/assets/60797147/19849b5d-20e6-4c5d-a7bc-785b5c06f638)

**Comment by tonylampada** [comment]
PR #162 ready for review.

**Comment by tonylampada** [comment]
FYI I just pushed a new version of the pip package ([roboflow-1.1.3](https://pypi.org/project/roboflow/)) that makes uploading large datasets more reliable.
One problem that happens sometimes (very rarely) when we upload data (images and/or annotations) is that the backend replies with a 502.
A 502 usually means an error  on google's infra: Google's load balancer for some reason didn't even call the function that was supposed to run the code for that request.
There's not much we can do about 502's except tell the user to try again.
Our pip package didn't know how to handle 502's and and if you got one in the middle of a large dataset upload, the upload would break and stop in the middle.
What's included in 1.1.3:
It handles 502 properly. If there's a 502 (or any other kind of error) with one image, it logs the error and moves on to the next one
Has a better separation between stdout and stderr. If the user redirects stderr to a file (with something like 2> stderr.log) they'll get only a list of the files that failed.
Fixes another bug where workspace.upload_dataset was creating a new project and uploading data to the new project everytime. Now, if the project name matches an existing project, it will upload data to the existing project.
Clients that were having problems with the pip package might want to update (pip install --upgrade roboflow) and try again
